students/chieu_quach/lesson07/html_render.py

```diff
@@ -0,0 +1,353 @@
+                self.tag = "li"
+            if self.tag == "pz":
+                self.tag = "p"
+
+            nameformat = "</{}>\n".format(self.tag)
+            if self.tag == "hr":
+          
+                out_file.write(indentation(nameformat, head_indent_txt))  
+           # out_file.write("<{} />\n".format(self.tag))
+           #   out_file.write(indentation(nameformat, head_indent_txt))
+            elif self.tag == "meta":
+                pass
+            elif self.tag == "head":
+         #   nameformat = "</{}>\n".format(self.tag)
+                out_file.write(indentation(nameformat, head_indent))
+            elif self.tag == "p":
+        #    nameformat = "</{}>\n".format(self.tag)
+                out_file.write(indentation(nameformat, head_indent_txt))
+            elif self.tag == "li":
+                out_file.write(indentation(nameformat, p_indent_txt))
+            elif self.tag == "ul":
+                out_file.write(indentation(nameformat, head_indent_txt))
+            elif self.tag == "body":
+                out_file.write(indentation(nameformat, head_indent))
+            else:
+            
+                out_file.write("</{}>\n".format(self.tag))
+
+
+
+            
+#     #   out_file.write("just something as a place holder...")
+
+class Meta(Element):
+    tag = "meta"
+
+class Html(Element):
+
+    tag = "html"
+  #  tag = "!DOCTYPE html"
+          
+    
+class Body(Element):
+    tag = "body"
+    
+class A(object):
+    tag = "a"
+    def __init__(self, link, content):
+    #    print ("self ", self)
+        if link is not None:
+            hrefmsg = """<a href="http://google.com", link</a> """
+            content = hrefmsg + "\n"
+       #     print(content)
+       #     print ("link ", link)
+
+        if content is not None:
+            self.contents = [content]
+
+    def append(self, new_content):
+        self.contents.append(new_content)
+        pass
+
+
+    def render(self, out_file):
+
+      #  if self.tag == "a":
+           
+         #   outfile.write("<{}>\n".format(self.tag))
+       #     print ("self tag ", self.tag)
+        for content in self.contents:
+           
+            try:                
+                content.render(out_file)
+            except AttributeError:           
+                out_file.write(content)
+
+class Li(Element):
+    tag = "li"
+                
+class Ul(Element):
+    tag = "ul"    
+
+
+
+class Head(Element):
+    tag = "head"
+
+
+    
+class Hr(Element):
+    tag = "hr"
+    
+class P(Element):
+    tag = "p"
+
+class PZ(Element):
+    tag = "pz"
+    
+class OneLineTag(Element):
+    
+    def render(self, out_file):
+        for content in self.contents:
+            
+            out_file.write("<{}>".format(self.tag))
+            out_file.write(self.contents[0])
+            out_file.write("</{}>\n".format(self.tag))
+
+      
+    def append(self, content):
+        raise NotImplementedError
+    
+class H(OneLineTag):
```

Comment:
"Create a Header class – this one should take an integer argument for the header level"

should just update the `tag` from the level variable and not other methods.